### ISSUE TITLE
rust-analyzer: "Function __wasm_bindgen_generated... should have snake_case name"

### ISSUE THREAD

**Issue by elmarco** [rustwasm/wasm-bindgen#4080]
### Describe the Bug

rust-analyzer complains about generated fn names.

### Steps to Reproduce

```rust
#[wasm_bindgen]
impl Session {
    pub fn desktop_size(&self) -> DesktopSize {
    ...
  }
}
```

Expands to
````rust
    pub fn desktop_size(&self) -> DesktopSize {
        #[automatically_derived]
        const _: () = {
            #[cfg_attr(
                all(target_arch = "wasm32", not(any(target_os = "emscripten", target_os = "wasi"))),
                export_name = "session_desktop_size"
            )]
            pub unsafe extern "C" fn __wasm_bindgen_generated_Session_desktop_size(
                me: u32,
            ) -> wasm_bindgen::convert::WasmRet<<DesktopSize as wasm_bindgen::convert::ReturnWasmAbi>::Abi>
````

### Expected Behavior

No rust-analyzer warning

### Actual Behavior

rust-analyzer warnings are emitted, such as

Function `__wasm_bindgen_generated_Session_desktop_size` should have snake_case name, e.g. `__wasm_bindgen_generated_session_desktop_size`

### Additional Context

Given that the fn have an export name, couldn't the fn symbol name be simply lowercased?

**Comment by daxpedda** [comment]
> Given that the fn have an export name, couldn't the fn symbol name be simply lowercased?

Probably not, otherwise a different `session::desktop_size()` would collide.

I was unable to reproduce this issue, so it would probably be best if you could make a small but complete reproducible example.

**Comment by gagbo** [comment]
I am having the issue with `__wbg_instanceof_JsType` from tsify ([the issue](https://github.com/madonoharu/tsify/issues/42)) and weirdly I wasn’t able to repro [in the playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=aa1f8364e8b248c91b0ce677db76c765) either.

Tsify triggers the warning from this generated code:
https://github.com/madonoharu/tsify/blob/66cddfe680ae429be0063cf6e6b1b5be34bb7ce2/tsify-macros/src/wasm_bindgen.rs#L57-L61

I think this commit https://github.com/rustwasm/wasm-bindgen/commit/ddceac747876d67fe52dafc90964f95741ae467d is the fix for this.

**Comment by daxpedda** [comment]
Thank you for looking into this!

If this is ultimately not fixed by #4078, I'm happy to take a look again with a reproducible example.
